### PR TITLE
[4.11] Fix set initial reservation on public IP ranges

### PR DESCRIPTION
**Pull request by nvazquez**
## Description
Fix initial reservation on public IP range addition. Also, if the range is dedicated for system VMs do not allow dedicating it to any account

Fixes: #2929 

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?
- Set reservation to a domain, add range. Verify domain is set to the domain.

### CODE REVIEW

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

File: server/src/com/cloud/configuration/ConfigurationManagerImpl.java [hunk 1]
```diff
@@ -3569,10 +3569,10 @@ public VlanVO doInTransaction(final TransactionStatus status) {
                     }
                     // increment resource count for dedicated public ip's
                     _resourceLimitMgr.incrementResourceCount(vlanOwner.getId(), ResourceType.public_ip, new Long(ips.size()));
-                } else if (domain != null) {
+                } else if (domain != null && !forSystemVms) {
                     // This VLAN is domain-wide, so create a DomainVlanMapVO entry
-                    //final DomainVlanMapVO domainVlanMapVO = new DomainVlanMapVO(domain.getId(), vlan.getId());
-                    //_domainVlanMapDao.persist(domainVlanMapVO);
+                    final DomainVlanMapVO domainVlanMapVO = new DomainVlanMapVO(domain.getId(), vlan.getId());
```
Comment (DaanHoogland) [inline server/src/com/cloud/configuration/ConfigurationManagerImpl.java]: the rest of the change seems trivial. I just wonder why these lines where commented in the first place. Can you explain @nvazquez ?

Comment (svenvogel) [reply server/src/com/cloud/configuration/ConfigurationManagerImpl.java]: as i can say. we tested it. it works now!

**Comment by svenvogel** [comment]
@nvazquez will it merged?

File: server/src/com/cloud/configuration/ConfigurationManagerImpl.java (hunk 1 above)
Comment (svenvogel) [reply server/src/com/cloud/configuration/ConfigurationManagerImpl.java]: @nvazquez is there any problem to merge it?

**Pull request merged**